src/prolog/lib/builtins.pl

```diff
@@ -933,13 +933,6 @@
     ;  throw(error(type_error(in_character, C), get_char/1))
     ).
 
-get_single_char(C) :-
-    (  var(C) -> '$get_single_char'(C)
-    ;  C == end_of_file  -> '$get_single_char'(C)
```

Comment:
Why is the `C == end_of_file` check here? It will cause `get_single_char(C)` to silently fail. I think `get_single_char` should throw `type_error(in_character, C)` in this case.

Reply:
I did a copy of `get_char/1`.

Reply:
I'm removing that check then.

Reply:
`in_character` is not a type! You can throw a `domain_error` on the grounds that the argument has the expected type (i.e., `atom`), but is not an element of the intended domain (i.e., one-char atoms).

Reply:
(I see this is what you did: Perfect!)